File: bindings/py/v1/my_lib/store.py
from ._generated import loader

class KVStore:
    def __init__(self):
        """Instantiates a simulated key-value store for Preview 1 (using Python state)."""
        self._data = {}

    def set(self, key: str, value: str) -> None:
        """Stores a key-value pair, calling WASM for processing."""
        loader.call("store.set", {"key": key, "value": value})
        self._data[key] = value

    def get(self, key: str) -> str | None:
        """Retrieves a value, calling WASM for processing."""
        loader.call("store.get", {"key": key})
        return self._data.get(key)

    def delete(self, key: str) -> bool:
        """Deletes a key, calling WASM for processing."""
        loader.call("store.delete", {"key": key})
        if key in self._data:
            del self._data[key]
            return True
        return False

File: bindings/my_lib/js/v2/my_lib/store.js
import { store } from './_generated/my_lib_component.js';

export class KVStore {
  constructor() {
    this._impl = new store.KvStore();
  }

  async set(key, value) {
    this._impl.set(key, value);
  }

  async get(key) {
    return this._impl.get(key);
  }

  async delete(key) {
    return this._impl.delete(key);
  }
}
